File: packages/device_info/lib/src/flutter_config/src/models/src/photo_optimization_result/photo_optimization_result.dart
import 'dart:typed_data';

class PhotoOptimizationResult {
  late int beforeSize;
  late int afterSize;
  late Uint8List optimizedImage;

  PhotoOptimizationResult({
    required this.beforeSize,
    required this.afterSize,
    required this.optimizedImage,
  });
}

File: packages/device_info/android/src/main/kotlin/com/rocket/device/info/models/PhotoOptimizationResult.kt
package com.rocket.device.info.models

data class PhotoOptimizationResult(
    val beforeSize: Long,
    val afterSize: Long,
    val optimizedImage: ByteArray,
) {
    fun toMap(): Map<String, Any>{
        return mapOf(
            "beforeSize" to beforeSize,
            "afterSize" to afterSize,
            "optimizedImage" to optimizedImage
        )
    }
}